src/sentry/web/frontend/project_settings.py

```diff
@@ -187,6 +200,24 @@ def clean_slug(self):
                                         'using that slug' % other.name)
         return slug
 
+    def clean_token(self):
+        token = self.cleaned_data.get('token')
+        if not token:
+            return
+        token_re = r'^[a-zA-Z0-9-]{1,40}$'
+        if not re.match(token_re, token):
+            raise forms.ValidationError('Invalid security token, must be: %s' % token_re)
+        return token
+
+    def clean_token_header(self):
+        token_header = self.cleaned_data.get('token_header')
+        if not token_header:
+            return
+        header_re = r'^[a-zA-Z0-9-]{1,20}$'
```

Comment:
These patterns are effectively arbitrary and likely over conservative, but should be sufficient for anyone using this and prevents someone putting in complete garbage.